modules/widgets/my-community.php

```diff
@@ -0,0 +1,240 @@
+				<?php _e( 'Include activity from likers', 'jetpack' ); ?>
+			</label>
+		</p>
+
+		<p>
+			<label for="<?php echo $this->get_field_id( 'include_followers' ); ?>">
+				<input type="checkbox" class="checkbox"  id="<?php echo $this->get_field_id( 'include_followers' ); ?>" name="<?php echo $this->get_field_name( 'include_followers' ); ?>" value="1" <?php checked( $include_followers, 1 ); ?> />
+				<?php _e( 'Include activity from followers', 'jetpack' ); ?>
+			</label>
+		</p>
+
+		<p>
+			<label for="<?php echo $this->get_field_id( 'include_commenters' ); ?>">
+				<input type="checkbox" class="checkbox"  id="<?php echo $this->get_field_id( 'include_commenters' ); ?>" name="<?php echo $this->get_field_name( 'include_commenters' ); ?>" value="1" <?php checked( $include_commenters, 1 ); ?> />
+				<?php _e( 'Include activity from commenters', 'jetpack' ); ?>
+			</label>
+		</p>
+
+		<?php
+	}
+
+	/**
+	 * Sanitize widget form values as they are saved.
+	 *
+	 * @see WP_Widget::update()
+	 *
+	 * @param array $new_instance Values just sent to be saved.
+	 * @param array $old_instance Previously saved values from database.
+	 *
+	 * @return array Updated safe values to be saved.
+	 */
+	function update( $new_instance, $old_instance ) {
+		$instance = array();
+		$instance['title'] = wp_kses( $new_instance['title'], array() );
+		if ( $instance['title'] === $this->default_title ) {
+			$instance['title'] = false; // Store as false in case of language change
+		}
+
+		$instance['number'] = (int) $new_instance['number'];
+		if ( !in_array( $instance['number'], array( 10, 50 ) ) )
+			$instance['number'] = 10;
+
+		$instance['include_likers']     = (bool) $new_instance['include_likers'];
+		$instance['include_followers']  = (bool) $new_instance['include_followers'];
+		$instance['include_commenters'] = (bool) $new_instance['include_commenters'];
+
+		delete_transient( "$this->id-{$instance['number']}" . (int) $instance['include_likers'] . (int) $instance['include_followers'] . (int) $instance['include_commenters'] . '-widget-data' );
+
+		return $instance;
+	}
+
+	/**
+	 * Front-end display of widget.
+	 *
+	 * @see WP_Widget::widget()
+	 *
+	 * @param array $args     Widget arguments.
+	 * @param array $instance Saved values from database.
+	 */
+	function widget( $args, $instance ) {
+		$title = isset( $instance['title' ] ) ? $instance['title'] : false;
+
+		if ( false === $title ) {
+			$title = $this->default_title;
+		}
+
+		$title = apply_filters( 'widget_title', $title );
+
+		echo $args['before_widget'];
+
+		if ( ! empty( $title ) ) {
+			echo $args['before_title'] . $title . $args['after_title'];
+		}
+
+		$transient_name = "$this->id-{$instance['number']}" . (int) $instance['include_likers'] . (int) $instance['include_followers'] . (int) $instance['include_commenters'] . '-widget-data';
```

Comment:
Alright! Since transients should be 40 characters or less in length, do you think we could shorten the widget name maybe (something like `jetpack_community`), and remove `widget` from the name? This way we wouldn't run into issues with sites with large Widget IDs.